concordium-node/src/configuration.rs

```diff
@@ -158,109 +188,146 @@ pub struct BakerConfig {
         long = "heap-profiling",
         help = "Profile the heap [(`cost`,-hc), (`type`, -hy), (`module`, -hm), (`description`, \
                 -hd)] in the Haskell subsystem",
-        default_value = "none"
+        default_value = "none",
+        env = "CONCORDIUM_NODE_BAKER_HEAP_PROFILING"
```

Comment:
I don't think this should have BAKER, it is misleading since it is not so much about being a baker, but rather about  runtime profiling.

Maybe this group should be
```
CONCORDIUM_NODE_RUNTIME_HEAP_PROFILING
```

Reply:
I agree. Should we rename the BakerConfig type to e.g. RuntimeConfig while we're at it?